mdjs-layout/src/MdjsLayout.js

```diff
@@ -0,0 +1,421 @@
+      }
+
+      .preview-story summary {
+        padding: 15px;
+        cursor: pointer;
+        border-top: 1px solid #9ca3af;
+        background-color: #f3f4f6;
+      }
+
+      .preview-story summary:focus {
+        outline: none;
+        border: 2px solid #404040;
+        padding: 14px 13px 13px;
+      }
+
+      .preview-story summary:hover {
+        background-color: #D4D4D4;
+      }
+
+      :host([theme='dark']) .preview-story summary {
+        background-color: #1b1e22;
+      }
+
+      :host([theme='dark']) .preview-story summary:hover {
+        background-color: #171717;
+      }
+
+      .preview-story pre[class*='language-'] {
+        margin: 0;
+      }
+
+      .preview-story details {
+        border-bottom-left-radius: 4px;
+        border-bottom-right-radius: 4px;
+        border-color: #9ca3af;
+        overflow: hidden;
+      }
+
+      .preview-story
+        details
+        > pre[class*='language-']
+        > code[class*='language-'] {
+        z-index: 0;
+      }
+
+      /**
+      *
+      *     DEFAULT PRISM THEME
+      *
+      */
+      pre[class*='language-'],
+      code[class*='language-'] {
+        color: #d4d4d4;
+        font-size: 13px;
+        text-shadow: none;
+        font-family: Menlo, Monaco, Consolas, 'Andale Mono', 'Ubuntu Mono',
+          'Courier New', monospace;
+        direction: ltr;
+        text-align: left;
+        white-space: pre;
+        word-spacing: normal;
+        word-break: normal;
+        line-height: 1.5;
+        -moz-tab-size: 4;
+        -o-tab-size: 4;
+        tab-size: 4;
+        -webkit-hyphens: none;
+        -moz-hyphens: none;
+        -ms-hyphens: none;
+        hyphens: none;
+      }
+
+      pre[class*='language-']::selection,
+      code[class*='language-']::selection,
+      pre[class*='language-'] *::selection,
+      code[class*='language-'] *::selection {
+        text-shadow: none;
+        background: #75a7ca;
+      }
+
+      @media print {
+        pre[class*='language-'],
+        code[class*='language-'] {
+          text-shadow: none;
+        }
+      }
+
+      pre[class*='language-'] {
+        padding: 1em;
+        margin: 0.5em 0;
+        overflow: auto;
+        background: #1e1e1e;
+      }
+
+      :not(pre) > code[class*='language-'] {
+        padding: 0.1em 0.3em;
+        border-radius: 0.3em;
+        color: #db4c69;
+        background: #f9f2f4;
+      }
+      /*********************************************************
+      * Tokens
+      */
+      .namespace {
+        opacity: 0.7;
+      }
+
+      .token.doctype .token.doctype-tag {
+        color: #569CD6;
+      }
+
+      .token.doctype .token.name {
+        color: #9cdcfe;
+      }
+
+      .token.comment,
+      .token.prolog {
+        color: #6a9955;
+      }
+
+      .token.punctuation,
+      .language-html .language-css .token.punctuation,
+      .language-html .language-javascript .token.punctuation {
+        color: #d4d4d4;
+      }

```

Comment:
I think we want to import styles from somewhere instead of hardcoding here, e.g. just from prism

Reply:
How do you propose to do this? Prism only publishes CSS output file, the only way I know how to import those and use in a component is by doing:

```js
import prismStyles from 'prismjs/themes/prism-dark.css' assert { type: 'css' };
```
This gives you a CSSStyleSheet (aka constructable stylesheet) which you can use in conjunction with Lit's static styles getter, but the support for import assertions is pretty much non-existent outside of Canary Chrome..

Reply:
I guess theoretically you could put this in the render method:
```html
<link rel="stylesheet" href="prismjs/themes/prism-dark.css">
```

but you will need to use an absolute URL because there's no way our build-steps are going to transform that bare path to the "real" output path, and it's pretty tricky to know what the right path will be in the context of mdjs-layout inside backlight docs.

I would rather have it hardcoded than rely on a path that will probably break a lot.

Reply:
this should be the way to go in the long run

```js
import prismStyles from 'prismjs/themes/prism-dark.css' assert { type: 'css' };
```

in the short term you might rely on the fact that Backlight will do the magic for you, e.g. `import prismStyles from 'prismjs/themes/prism-dark.css'`, and then you can release a dockit package with transpiled code which should be compatible with any DS... I hope my assumption is right, maybe just play with it and see how it works

and the least good alternative is too use an absolute path to a CDN

Reply:
I couldn't get that to work, mostly because CSS imports (magic ones) in backlight just seems to insert the CSS in the <head>, but since the mdjs-layout has a shadow boundary, I need it inserted in the shadowDOM instead. Maybe you have an idea to hack it in some other way?

Reply:
https://backlight.dev/docs/language-css#import-as-a-cssstylesheet

Reply:
Maybe useless I would say sorry

Reply:
![image](https://user-images.githubusercontent.com/36734656/132241740-3a960bef-9ac4-4e43-beb8-6436ef309591.png)

Tried that but got an error, seems like it's trying to import as a JS module instead of a CSSStyleSheet

Reply:
my first impression after debugging this is that `processCssImport` is broken for CSS imports from absolute URLs, as well as from bare module specifiers, you can have a look if you want how to extend it to support this scenario, but maybe indeed just use `<link rel="stylesheet" href="https://unpkg.com/browse/prismjs@1.24.1/themes/prism-dark.css` for now, I checked and it works

Reply:
one thing I noticed now and don't get is why you decided to put everything in the Shadow DOM and no just the demo

I think we aligned earlier that the most important encapsulation is needed for the demo code itself... and encapsulation for the code block and some border around the demos is most likely not needed

maybe I'm wrong, let's have a chat tomorrow again about what and why we need to encapsulate, to me it's the most important decision here, so I wanna be confident that all requirements are taken into account

Reply:
Ah true, not sure why I thought we needed the encapsulation for the page wrapper. I updated the PR, without encapsulation it makes the styling (prism) easier too